plugins/amqp/amqp_input.go

```diff
@@ -203,7 +203,8 @@ func (ai *AMQPInput) Run(ir InputRunner, h PluginHelper) (err error) {
 		}
 		msg.Ack(false)
 	}
-	return nil
+	// return an error message to trigger a potential restart of the plugin
+	return fmt.Errorf("Channel closed while reading from queue %s",  ai.config.Queue)
```

Comment:
That's fine, but the purist in me is still a bit put off by returning an error when there isn't really an error case. I'll add some code so the `Stop` method sets a flag, which the `Run` method can check on the way out. If the flag is set, we'll return nil; if not, we return the error.